scripts/genCurrencies.mjs

```diff
@@ -20,54 +22,103 @@ import { getCurrencyData } from './getCurrencyData.mjs';
 // eslint-disable-next-line import/no-named-as-default-member
 const { format, resolveConfig } = prettier;
 
+const baseOutputDir = path.resolve(
+  url.fileURLToPath(new URL('.', import.meta.url)), // __dirname es module style
+  '../packages/currencies/src/iso4217/amendments/168'
+);
+
 const dirExists = (dirPath) =>
   fs
     .stat(dirPath)
     .then((stats) => stats.isDirectory())
     .catch(() => false);
 
+/** @type {import("./genCurrencies").CreateCurrencyConfig} */
+function createCurrencyConfig({ genericType, imports, mapNumber }) {
+  return {
+    genericType,
+    generateFn: ({ code, description, ...currency }) => `
+      import type { Currency } from '../../../../types';
+      ${imports ? imports.map((text) => `${text}\n`) : ''}
+      /**
+       * ${description}.
+       */
+      export const ${code}: Currency<${genericType}> = {
+        code: '${code}',
+        base: ${mapNumber(currency.base)},
+        exponent: ${mapNumber(currency.exponent)},
+      };`,
+  };
+}
+
+const numberCurrencyConfig = createCurrencyConfig({
+  genericType: 'number',
+  mapNumber: (n) => `${n}`,
+});
+
+const bigintCurrencyConfig = createCurrencyConfig({
+  genericType: 'bigint',
+  mapNumber: (n) => `${n}n`,
+});
+
 /**
  * Generate a set of currency source files.
  *
- * @param {string} outputDir - The directory to output the source files into.
- * @param {formatFn} formatFn - The format function used to generate individual currency files.
  * @returns {Promise<void>} - Returns a Promise that resolves when all the files are generated.
  */
-export async function genCurrencies(outputDir, formatFn) {
+export async function build() {
   const currencies = await getCurrencyData();
 
-  if (!(await dirExists(outputDir))) {
-    await fs.mkdir(outputDir, { recursive: true });
-  }
-
   const prettierOptions = {
     ...(await resolveConfig(process.cwd())),
     parser: 'babel',
   };
 
+  const configs = [numberCurrencyConfig, bigintCurrencyConfig];
+
+  // Generate index ts files
+  const indexPromises = configs.map(async ({ genericType }) => {
+    const outputDir = path.join(baseOutputDir, genericType);
```

Comment:
I'm using the name of the `genericType` to create the folder structure. I don't know if this is best, but it works for these two types.